rust/src/client.rs

```diff
@@ -14,397 +14,311 @@ use tracing::instrument;
 #[derive(Parser, Debug, Serialize, Deserialize)]
 #[command(author, version, about, long_about = None)]
 pub enum ClientCli {
-    /// Query account by public key
-    Account(AccountArgs),
-    #[command(subcommand)]
-    Block(BlockArgs),
-    #[command(subcommand)]
-    Chain(ChainArgs),
-    /// Create a checkpoint of the indexer store
-    Checkpoint(CheckpointArgs),
-    #[command(subcommand)]
-    Ledger(LedgerArgs),
+    #[clap(subcommand)]
+    Accounts(Accounts),
+    #[clap(subcommand)]
+    Blocks(Blocks),
+    #[clap(subcommand)]
+    Chain(Chain),
+    #[clap(subcommand)]
+    Checkpoints(Checkpoints),
+    #[clap(subcommand)]
+    Ledger(Ledger),
     #[command(subcommand)]
-    StakingLedger(StakingLedgerArgs),
+    StakingLedger(StakingLedger),
     #[clap(hide = true)]
     Shutdown,
     #[clap(subcommand)]
-    Snark(SnarkArgs),
+    Snark(Snark),
     /// Show a summary of the state
-    Summary(SummaryArgs),
-    #[command(subcommand)]
-    Transactions(TransactionArgs),
-    #[clap(flatten)]
-    InternalCommand(InternalCommandArgs),
+    Summary {
+        /// Path to write the summary [default: stdout]
+        #[arg(short, long)]
+        path: Option<PathBuf>,
+        /// Verbose output should be redirected to a file
+        #[arg(short, long, default_value_t = false)]
+        verbose: bool,
+        /// Output JSON data
+        #[arg(short, long, default_value_t = false)]
+        json: bool,
+    },
+    #[clap(subcommand)]
+    Transactions(Transactions),
+    #[clap(subcommand)]
+    InternalCommands(InternalCommands),
```

Comment:
Should we be consistent with making command names plural vs. singular?

Reply:
I think we should be consistent.

Can we use the following or are these really singular?

- StakingLedgers?
- Ledgers?
- Snarks?